k8srm-prototype/pkg/api/claim_types.go

```diff
@@ -355,16 +298,76 @@ type DeviceConfigReference struct {
 	Name string `json:"name"`
 }
 
-// DevicePoolAllocation contains the pool and number of selected devices.
-type DevicePoolAllocation struct {
+// DeviceAllocation contains an individual device allocation result, including
+// per-device resource allocations, when applicable.
+type DeviceAllocation struct {
+	// DevicePoolName is the name of the DevicePool to which this
+	// device belongs.
+	// +required
+	DevicePoolName string `json:"devicePoolName"`
```

Comment:
I am concerned DeviceName will not be unique across drivers. Since pools are managed by drivers, I guess I was thinking this would ensure uniqueness of the name (as well as make finding the device easy).

We could add driver here instead?